### PR TITLE
feat: get image info in the render function of `Image` component.  

### LINKED ISSUE #48713: 图片预览时为什么不会返回当前图片信息？
### What problem does this feature solve?

图片与视频实时切换

### What does the proposed API look like?

比如在toolbarRender方法中返回对应图片信息，可根据当前图片状态或者类型来渲染的对应的功能操作按钮

### PR DESCRIPTION
**Pull request by nova1751**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #48713

### 💡 Background and solution

Add imgInfo for render function.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Gets image info in the render function of `Image` component.       |
| 🇨🇳 Chinese |    在 `Image` 组件渲染函数中获取图片的信息。      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by nova1751** [comment]
Please check this PR https://github.com/react-component/image/pull/336 first, most of the code change is in `rc-image` repo.

File: package.json [hunk 1]
```diff
@@ -171,7 +171,7 @@
     "@ant-design/tools": "^18.0.2",
     "@antv/g6": "^4.8.24",
     "@babel/eslint-plugin": "^7.24.5",
-    "@biomejs/biome": "^1.7.1",
+    "@biomejs/biome": "^1.7.2",
```
Comment (yoyo837) [inline package.json:174]: dev的版本号升级不应该包含在当前PR

**Comment by yoyo837** [comment]
请根据提示解决CI报错，添加测试用例覆盖。

**Comment by afc163** [comment]
https://github.com/react-component/image/releases/tag/v7.7.0

**Comment by afc163** [comment]
CI broken, please fix.

**Comment by nova1751** [comment]
> CI broken, please fix.

Get it. Seems to be snapshot problem.

File: components/image/index.zh-CN.md [hunk 2]
```diff
@@ -97,7 +98,7 @@ coverDark: https://mdn.alipayobjects.com/huamei_7uahnr/afts/img/A*LVQ3R5JjjJEAAA
 | forceRender | 强制渲染预览图 | boolean | - | - |
 | countRender | 自定义预览计数内容 | (current: number, total: number) => React.ReactNode | - | 4.20.0 |
 | toolbarRender | 自定义工具栏 | (originalNode: React.ReactElement, info: [ToolbarRenderInfoType](#toolbarrenderinfotype)) => React.ReactNode | - | 5.7.0 |
-| imageRender | 自定义预览内容 | (originalNode: React.ReactElement, info: { transform: [TransformType](#transformtype), current: number }) => React.ReactNode | - | 5.7.0 |
+| imageRender | 自定义预览内容 | (originalNode: React.ReactElement, info: { transform: [TransformType](#transformtype), image: [ImgInfo](#imginfo), current: number }) => React.ReactNode | - | 5.7.0 |
```
Comment (li-jia-nan) [inline components/image/index.zh-CN.md:101]:
```suggestion
| imageRender | 自定义预览内容 | (originalNode: React.ReactElement, info: { transform: [TransformType](#transformtype), image: [ImgInfo](#imginfo), current: number }) => React.ReactNode | - | 5.7.0, imginfo: 5.18.0 |
```

File: components/image/index.en-US.md [hunk 3]
```diff
@@ -96,7 +97,7 @@ Other attributes [&lt;img>](https://developer.mozilla.org/en-US/docs/Web/HTML/El
 | forceRender | Force render preview dialog | boolean | - | - |
 | countRender | Custom preview count content | (current: number, total: number) => React.ReactNode | - | 4.20.0 |
 | toolbarRender | Custom toolbar render | (originalNode: React.ReactElement, info: [ToolbarRenderInfoType](#toolbarrenderinfotype)) => React.ReactNode | - | 5.7.0 |
-| imageRender | Custom preview content | (originalNode: React.ReactElement, info: { transform: [TransformType](#transformtype), current: number }) => React.ReactNode | - | 5.7.0 |
+| imageRender | Custom preview content | (originalNode: React.ReactElement, info: { transform: [TransformType](#transformtype), image: [ImgInfo](#imginfo), current: number }) => React.ReactNode | - | 5.7.0 |
```
Comment (li-jia-nan) [inline components/image/index.en-US.md:100]:
```suggestion
| imageRender | Custom preview content | (originalNode: React.ReactElement, info: { transform: [TransformType](#transformtype), image: [ImgInfo](#imginfo), current: number }) => React.ReactNode | - | 5.7.0, imginfo: 5.18.0 |
```

File: components/image/demo/preview-imgInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+const src = 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png';
+
+const App: React.FC = () => (
+  <Image
+    src={src}
+    width="200px"
+    height="200px"
+    alt="test"
+    preview={{
+      imageRender(_, { image }) {
```
Comment (li-jia-nan) [inline components/image/demo/preview-imgInfo.tsx:13]: 感觉两个重复的方法可以提出去：

```tsx
const src = 'https://zos.alipayobjects.com/rmsportal/jkjgkEfvpUPVyRjUImniVslZfWPnJuuZ.png';

const renderNode = (_: xxxx, info: xxxx) => (
  <div>{JSON.stringify(info.image, null, 2)}</div>
);

const App: React.FC = () => (
  <Image
    src={src}
    width="200px"
    height="200px"
    alt="test"
    preview={{ imageRender: renderNode, toolbarRender: renderNode }}
  />
);

export default App;
```

Comment (nova1751) [reply components/image/demo/preview-imgInfo.tsx:13]: 两个函数功能还是不一样考虑到以后有更改的需要，还是分开比较好吧。

Comment (li-jia-nan) [reply components/image/demo/preview-imgInfo.tsx:13]: 有道理，那改成箭头函数吧，一行就可以写完了

**Review by afc163 — APPROVED** [review]

**Comment by zhu0629** [comment]
5.18更新后，图片预览时偶尔会出现默认显示很小的尺寸。请检查。

**Pull request merged**